modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorage.java

```diff
@@ -196,6 +223,41 @@ public long getCompactionRevision() {
         }
     }
 
+    @Override
+    public void startCompaction(long revision) {
+        assert revision >= 0 : revision;
+
+        rwLock.writeLock().lock();
+
+        try {
+            assertCompactionRevisionLessThanCurrent(revision, rev);
+
+            if (isRecoveryState()) {
+                compactionRevision = revision;
+            } else {
+                watchProcessor
+                        .addTaskToWatchEventQueue(() -> setCompactionRevision(revision))
+                        .thenComposeAsync(unused -> readOperationsFuture(revision), compactionExecutor)
+                        .thenRunAsync(() -> compact(revision), compactionExecutor)
+                        .whenComplete((unused, throwable) -> {
```

Comment:
I would use `whenCompleteAsync` here, just in case

Reply:
Why not.